portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -67,7 +72,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   private String convertToJson(List<Entity> messages) {
     String json = "{\"history\": [";
     for (int i = 0; i < messages.size(); i++) {
-      json += "\"" + (String) messages.get(i).getProperty("content") + "\"";
+      Entity cur = messages.get(i);
+      json += "{";
+      json += "\"" + COMM_NAME + "\"" + ": \"" + (String) cur.getProperty(COMM_NAME) + "\"" + ", ";
+      json += "\"" + COMM_CONTENT + "\"" + ": \"" + (String) cur.getProperty(COMM_CONTENT) + "\"";
+      json += "}";
       if (i != messages.size() - 1) {
         json += ", ";
```

Comment:
will you get an error if this extra comma is included?

Reply:
replaced convertToJson function with Gson